File: golanglang/internal/binary-search/array-binary-search.go
package binarysearch

func ArrayBinarySearch(list []int, target int) int {
	lowestIndex := 0
	highestIndex := len(list) - 1

	for lowestIndex <= highestIndex {
		middleIndex := (lowestIndex + highestIndex) / 2
		middleItem := list[middleIndex]

		if middleItem == target {
			return middleIndex
		}

		if middleItem > target {
			highestIndex = middleIndex - 1
		} else {
			lowestIndex = middleIndex + 1
		}
	}

	return -1
}

File: tslang/src/array/binary-search/ArrayBinarySearch.ts
const arrayBinarySearch = (list: number[], target: number) => {
	let highestIndex = list.length - 1;
	let lowestIndex = 0;

	while (lowestIndex <= highestIndex) {
		const middleIndex = Math.round((lowestIndex + highestIndex) / 2);
		const middleItem = list[middleIndex];

		if (target === middleItem) {
			return middleIndex;
		}

		if (middleItem > target) {
			highestIndex = middleIndex - 1;
		} else {
			lowestIndex = middleIndex + 1;
		}
	}

	return -1;
};

export { arrayBinarySearch };
